### PR TITLE
Anticipated load from frontend

### PR DESCRIPTION
**Pull request by burnout87**
Loading of list files and catalog files from front-end is now performed before `job_id` calculation, and also before the scratch folder creation.

The data have therefore to be saved temporarily in another folder.

### CODE REVIEW

**Comment by volodymyrss** [comment]
> This pull request **introduces 17 alerts** and **fixes 1** when merging [9e44f77](https://github.com/oda-hub/dispatcher-app/commit/9e44f77a339c74992be0d31d34af3e08fd5072a5) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-f7a91240bf1a90049bb73acbbd1c4342e71d3d6c)
> 
> **new alerts:**
> 
>     * 17 for Module-level cyclic import

hm, that's unfortunate, may cause issues

**Comment by burnout87** [comment]
> > This pull request **introduces 17 alerts** and **fixes 1** when merging [9e44f77](https://github.com/oda-hub/dispatcher-app/commit/9e44f77a339c74992be0d31d34af3e08fd5072a5) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-f7a91240bf1a90049bb73acbbd1c4342e71d3d6c)
> > **new alerts:**
> > ```
> > * 17 for Module-level cyclic import
> > ```
> 
> hm, that's unfortunate, may cause issues

It's the first time I see it.

**Comment by volodymyrss** [comment]
> > > This pull request **introduces 17 alerts** and **fixes 1** when merging [9e44f77](https://github.com/oda-hub/dispatcher-app/commit/9e44f77a339c74992be0d31d34af3e08fd5072a5) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-f7a91240bf1a90049bb73acbbd1c4342e71d3d6c)
> > > **new alerts:**
> > > ```
> > > * 17 for Module-level cyclic import
> > > ```
> > 
> > 
> > hm, that's unfortunate, may cause issues
> 
> It's the first time I see it.

Do you mean, ever?
In this project?

**Comment by burnout87** [comment]
> > > > This pull request **introduces 17 alerts** and **fixes 1** when merging [9e44f77](https://github.com/oda-hub/dispatcher-app/commit/9e44f77a339c74992be0d31d34af3e08fd5072a5) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-f7a91240bf1a90049bb73acbbd1c4342e71d3d6c)
> > > > **new alerts:**
> > > > ```
> > > > * 17 for Module-level cyclic import
> > > > ```
> > > 
> > > 
> > > hm, that's unfortunate, may cause issues
> > 
> > 
> > It's the first time I see it.
> 
> Do you mean, ever?
> In this project?

I mean that I am trying to understand what caused this issue with the changes that I applied. And also in this project yes, I haven't seen it yet

**Comment by volodymyrss** [comment]
> > > > > This pull request **introduces 17 alerts** and **fixes 1** when merging [9e44f77](https://github.com/oda-hub/dispatcher-app/commit/9e44f77a339c74992be0d31d34af3e08fd5072a5) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-f7a91240bf1a90049bb73acbbd1c4342e71d3d6c)
> > > > > **new alerts:**
> > > > > ```
> > > > > * 17 for Module-level cyclic import
> > > > > ```
> > > > 
> > > > 
> > > > hm, that's unfortunate, may cause issues
> > > 
> > > 
> > > It's the first time I see it.
> > 
> > 
> > Do you mean, ever?
> > In this project?
> 
> I mean that I am trying to understand what caused this issue with the changes that I applied. 

Indeed, unclear.

> And also in this project yes, I haven't seen it yet

it's because it was just introduced!

File: cdci_data_analysis/flask_app/dispatcher_query.py [hunk 1]
```diff
@@ -160,6 +160,20 @@ def __init__(self, app, instrument_name=None, par_dic=None, config=None, data_se
             else:
                 logger.debug("NOT get_meta_data request: yes scratch_dir")
 
+                self.set_sentry_client()
+                if data_server_call_back is False:
```
Comment (volodymyrss) [inline cdci_data_analysis/flask_app/dispatcher_query.py:164]: at least pep8 does not encourage this bool comparison https://www.python.org/dev/peps/pep-0008/

**Review by volodymyrss — COMMENTED** [review]
I see the complexity; I guess it is good. Though I wonder if it is possible to avoid writing the file at all, and deal with it in-memory.

**Comment by burnout87** [comment]
> I see the complexity; I guess it is good. Though I wonder if it is possible to avoid writing the file at all, and deal with it in-memory.

Probably yes, I just didn't investigate this option since it would probably take even more time. It is though a good future improvement.

**Review by volodymyrss — APPROVED** [review]

**Comment by burnout87** [comment]
> This pull request **fixes 6 alerts** when merging [f59ef33](https://github.com/oda-hub/dispatcher-app/commit/f59ef332363fc5a2dfc84a1b0460624faeba0f95) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-4afb885e5423e6eb4682f9b951a81ce5568a50ae)
> 
> **fixed alerts:**
> 
>     * 6 for Unused import

They no disappeared?

**Comment by volodymyrss** [comment]
> > This pull request **fixes 6 alerts** when merging [f59ef33](https://github.com/oda-hub/dispatcher-app/commit/f59ef332363fc5a2dfc84a1b0460624faeba0f95) into [4d30005](https://github.com/oda-hub/dispatcher-app/commit/4d30005380b5bfd6c7dbeab818b282335db8af35) - [view on LGTM.com](https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-4afb885e5423e6eb4682f9b951a81ce5568a50ae)
> > **fixed alerts:**
> > ```
> > * 6 for Unused import
> > ```
> 
> They no disappeared?

seems so

you can see the whole issue set?

https://lgtm.com/projects/g/oda-hub/dispatcher-app/rev/pr-4afb885e5423e6eb4682f9b951a81ce5568a50ae

**Pull request merged**